test/types/schema_helpers.test-d.ts

```diff
@@ -1,18 +1,22 @@
 import { Document, ObjectId } from 'bson';
-import { expectAssignable, expectNotType, expectType } from 'tsd';
+import { expectAssignable, expectError, expectNotType, expectType } from 'tsd';
 
 import type {
   EnhancedOmit,
   InferIdType,
   OptionalId,
+  OptionalUnlessRequiredId,
   WithId,
   WithoutId
 } from '../../src/mongo_types';
 
+declare function optionalReturnValue(): number | ObjectId;
 // InferIdType
 expectType<InferIdType<Document>>(new ObjectId());
 expectType<InferIdType<{ _id: number }>>(1 + 1);
 expectType<InferIdType<{ a: number } | { b: string }>>(new ObjectId());
+expectType<InferIdType<{ _id?: number | ObjectId }>>(optionalReturnValue());
```

Comment:
should this test be just that infer id returns number for an optional _id? (i.e., `InferIdType<{ _id?: number }>`)